docs(docs): Initial docset & expanded README (#14)
spinnaker/rotation-scheduler: README.md

> # rotation-scheduler

Reviewer: I know the actual repo name needs to be hyphenated, but in common usage could this be called "Rotation Scheduler"?

Author: Done.

> A GitHub Action for generation a rotation schedule.

Reviewer: ```suggestion
A GitHub Action to generate a rotation schedule.
```

Author: Done.

- startDate: Sun 05 Apr 2020
  user: xyz
- startDate: Sun 12 Apr 2020
  user: "123"
- startDate: Sun 19 Apr 2020
  stopDate: Sat 25 Apr 2020
  user: abc
```

Use `--prune` flag to remove past shifts and reschedule shifts when users drop out of rotation:
```bash
$ date '+%Y-%m-%d' # pruning works from the current date. Only whole shifts are removed.
2020-04-09 

$ rotation schedule extend --prune --schedule rotation-schedule.yaml --stop 2020-05-01 --users lmn,xyz,123 rotation-schedule.yaml
$ cat rotation-schedule.yaml 
shifts:
- startDate: Sun 05 Apr 2020
  user: xyz
- startDate: Sun 12 Apr 2020
  user: "123"
- startDate: Sun 19 Apr 2020
  stopDate: Sat 25 Apr 2020
  user: lmn
```

Includes GitHub Teams integration:
```bash
$ rotation schedule generate --start 2020-03-01 --stop 2020-04-01 --github spinnaker,build-cops,$GITHUB_TOKEN
shifts:
- startDate: Sun 01 Mar 2020
  user: ajordens
- startDate: Sun 08 Mar 2020
  user: cfieber
- startDate: Sun 15 Mar 2020
  user: ethanfrogers
- startDate: Sun 22 Mar 2020
  stopDate: Sat 28 Mar 2020
  user: ezimanyi
```

Sync schedule to Google Calendar. If `user` fields are email addresses, the email address is added as an event attendee.

> The `--calendarID` **must** be a G-Suite user account dedicated for this purpose - all events are cleared during a sync!

Reviewer: ```suggestion
The `--calendarID` **must** be a G Suite user account dedicated for this purpose - all events are cleared during a sync!
```

Author: Done.

> The `--calendarID` **must** be a G Suite user account dedicated for this purpose - all events are cleared during a sync!

Reviewer: ```suggestion
The `--calendarID` **must** be a G Suite user account dedicated for this purpose. All events are cleared during a sync!
```

Reviewer: Missed this before. What does it mean that all events are cleared during a sync?

Author: All events on that users's calendar are wiped clean (deleted) first, and then the new schedule is put in place. It was the easiest sync logic I could come up with.
